src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1093,7 +1093,7 @@ internal bool MethodGroupIsCompatibleWithDelegate(BoundExpression receiverOpt, b
                 if (!CheckConstantBounds(destinationType, sourceValue))
                 {
                     // NOTE: Dev10 puts a suffix, "M", on the constant value.
-                    Error(diagnostics, ErrorCode.ERR_ConstOutOfRange, syntax, sourceValue.Value + "M", destination);
+                    Error(diagnostics, ErrorCode.ERR_ConstOutOfRange, syntax, sourceValue.Value + "M", destination!);
```

Comment:
>destination [](start = 103, length = 11)

Can't `destination` be null here? Seems dangerous

Reply:
The null check above is actually superfluous.

---
In reply to: [367184489](https://github.com/dotnet/roslyn/pull/40946#discussion_r367184489) [](ancestors = 367184489)

Reply:
Sorry, I'd missed your reply. Consider removing the superfluous null check so that we don't need those suppressions.
Suppressions are typically for when the compiler is not able to analyze properly, but here it's rather that the code is inconsistent (checks for null in one place, but would blow up with null in another).

---
In reply to: [370290111](https://github.com/dotnet/roslyn/pull/40946#discussion_r370290111) [](ancestors = 370290111,367184489)

Reply:
I'm not going to change the semantics of the code in this PR.

---
In reply to: [375590101](https://github.com/dotnet/roslyn/pull/40946#discussion_r375590101) [](ancestors = 375590101,370290111,367184489)